mathesar/serializers.py

```diff
@@ -24,6 +24,15 @@ class Meta:
         model = Schema
         fields = ['id', 'name', 'database', 'tables']
 
+    def validate_name(self, name):
+        database = self.initial_data.get('database', '')
+        # Can't use .filter() since 'name' is a property of Schema, not a field
+        existing = [schema for schema in Schema.objects.all()
```

Comment:
I think it would be better to use the list of schema names from the `db` module here, since the database is the source of truth for schema names.